### PR TITLE
Replace EXPECT_TRUE where sensible

### PR DESCRIPTION
**Pull request by masterleinad**
While looking at #4887, I noted all the instances of using `ASSERT_TRUE` where we have more suitable replacements that print more information upon failure.

### CODE REVIEW

File: core/unit_test/TestResize.hpp [hunk 1]
```diff
@@ -382,7 +382,7 @@ void impl_testResize() {
         }
       }
     }
-    EXPECT_TRUE(test == true);
+    EXPECT_EQ(test, true);
```
Comment (dalg24) [inline core/unit_test/TestResize.hpp:385]:
```suggestion
    EXPECT_TRUE(test);
```

File: core/unit_test/TestResize.hpp [hunk 2]
```diff
@@ -344,7 +344,7 @@ void impl_testResize() {
...
```
Comment (dalg24) [inline core/unit_test/TestResize.hpp:347]: They are more occurrences

Comment (masterleinad) [reply core/unit_test/TestResize.hpp]: See [`d2072d0` (#4888)](https://github.com/kokkos/kokkos/pull/4888/commits/d2072d0b92447d08fd958bec1206302fb0f34af1).

**Review by crtrott — APPROVED** [review]

File: algorithms/unit_tests/TestStdAlgorithmsCopyIf.cpp [hunk 3]
```diff
@@ -220,13 +220,13 @@ void verify_data(const std::string& name, ViewTypeFrom view_from,
     std::size_t count = 0;
     for (std::size_t i = 0; i < view_from_h.extent(0); ++i) {
       if (pred(view_from_h(i))) {
-        EXPECT_TRUE(view_test_h(count++) == view_from_h(i));
+        EXPECT_EQ(view_test_h(count++), view_from_h(i));
```
Comment (nliber) [inline algorithms/unit_tests/TestStdAlgorithmsCopyIf.cpp:223]: I know this is in the original code, but it is bad to have side effect code (`count++`) inside macros.

(Unless I missed another one, this appears to be the only occurrence.)

File: containers/unit_tests/TestScatterView.hpp [hunk 4]
```diff
@@ -259,11 +259,11 @@ struct test_scatter_view_impl_cls<DeviceType, Layout, Duplication, Contribution,
       auto val0 = host_view(i, 0);
       auto val1 = host_view(i, 1);
       auto val2 = host_view(i, 2);
-      EXPECT_TRUE(std::fabs((val0 - 65536.0) / 65536.0) < 1e-14)
+      EXPECT_LT(std::fabs((val0 - 65536.0) / 65536.0), 1e-14)
```
Comment (nliber) [inline containers/unit_tests/TestScatterView.hpp:262]: I think `EXPECT_NEAR` would be more appropriate in these cases.  [GoogleTest Assertions](https://google.github.io/googletest/reference/assertions.html)

**Review by nliber — CHANGES_REQUESTED** [review]

File: algorithms/unit_tests/TestStdAlgorithmsCopyIf.cpp (hunk 3 above)
Comment (masterleinad) [reply algorithms/unit_tests/TestStdAlgorithmsCopyIf.cpp]: 05d532023a65680ef0bf0c86a22138cfbf94e334

File: containers/unit_tests/TestScatterView.hpp (hunk 4 above)
Comment (masterleinad) [reply containers/unit_tests/TestScatterView.hpp]: 6684cda8b4f826e7913b77a475a1877cc3a8aa9e

**Comment by masterleinad** [comment]
`OPENMPTARGET-Clang` is failing with an ICE which seems unrelated.

**Review by nliber — APPROVED** [review]

**Pull request merged**